### ISSUE TITLE
zipcmp: malloc "pointer being freed was not allocated" error on macOS

### ISSUE THREAD

**Issue by samford** [nih-at/libzip#150]
**Describe the Bug**
In the process of upgrading libzip to v1.6.0 in the Homebrew package manager (Homebrew/homebrew-core#49386), I found that `zipcmp` encountered a runtime error on macOS. libzip builds fine and `zipmerge` and `ziptool` seem to work properly, so it's only `zipcmp` that has a problem.  This issue occurred on the tested macOS versions (10.13-10.15) and was newly introduced in libzip v1.6.0 (i.e., v1.5.2 does not have this issue under the same circumstances).

**Expected Behavior**
`zipcmp` should produce normal output without runtime errors, for example:

```
$ zipcmp file1.zip file2.zip
--- file1.zip
+++ file2.zip
+         18 f9c5c01d file2
```

**Observed Behavior**
The expected output is displayed but is followed by `malloc` errors:

```
$ zipcmp file1.zip file2.zip
--- file1.zip
+++ file2.zip
+         18 f9c5c01d file2
zipcmp(88074,0x112f51dc0) malloc: *** error for object 0x67efcc51328bba77: pointer being freed was not allocated
zipcmp(88074,0x112f51dc0) malloc: *** set a breakpoint in malloc_error_break to debug
```

**To Reproduce**
Build libzip on macOS using the usual [installation instructions](https://github.com/nih-at/libzip/blob/master/INSTALL.md) and run `zipcmp` in a normal fashion.

**libzip Version**
1.6.0

**Operating System**
macOS 10.13, 10.14, 10.15 (potentially other versions as well).  I tested this locally with macOS 10.15.2 and 10.15.3, using AppleClang 11.0.0.11000033 and cmake 3.16.2.

**Test Files**
This issue doesn't seem to require any special zip files to encounter it.

**Additional context**
Nothing to note but let me know if you need further information.

**Comment by 0-wiz-0** [comment]
Thanks. I had already found this shortly after the release (sadly, not before) and fixed it in bd9fb11fa88f6a95b8afbd53796934adf4e6e6bd.

**Comment by fxcoudert** [comment]
Could you make a point release to fix the issue? It seems pretty major

**Comment by 0-wiz-0** [comment]
I've just released 1.6.1.
If you don't use `zipcmp`, you don't need to update.
